### PR TITLE
Update PUT list subscription request params with appropriate type

### PR DESCRIPTION
**Pull request by tk26**
## Description of the change

> Adds types for PUT list subscriptions' request params

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@trycourier/courier",
-  "version": "1.7.1",
+  "version": "1.7.2",
```
Comment (aydrian) [inline package.json:3]: If we are going to bump the version, please also include updates to the ChangeLog so we can push a new tag after this has been merged.

**Review by aydrian — CHANGES_REQUESTED** [review]
LGTM. Just need the Change Log updates for the v1.7.2 release.

File: package.json (hunk 1 above)
Comment (tk26) [reply package.json:3]: Good point. Done!

**Review by aydrian — APPROVED** [review]
LGTM :shipit:

**Pull request merged**